src/custom-json-channel.js

```diff
@@ -1,6 +1,73 @@
 const axios = require("axios");
 const {findAppropriateUrl} = require("./methods_info");
 
+/**
+ * This recursively goes through the object/array and serialized any "base64" field that has Buffer value to a BASE64
+ * string.
+ */
+function stringifyBase64(obj) {
+    if (obj.constructor === Array) {
+        return obj.map(stringifyBase64);
+    } else if (obj.constructor === Object) {
+        const processedObject = {}
+        for (const key in obj) {
+            if (obj.hasOwnProperty(key)) {
+                if (key === "base64") {
+                    if (Buffer.isBuffer(obj[key])) {
+                        processedObject[key] = obj[key].toString("base64");
```

Comment:
Isn’t a buffer the wrong type for a byes field? Isn’t this a user error if they set a buffer? This would send the wrong value if they used the binary channel. We should let grpc decide what to send like in python. When their included json format is used it will convert bytes fields to base64 since json can’t handle them otherwise. But when using the binary channel it doesn’t do that conversion and keeps the bytes as is. Is there no included json format they provide for this language?

Reply:
@zeiler The default gRPC channel supports setting the `bytes base64` field to either "Buffer" or "string" types. If you set "Buffer", I assume it will automatically send the bytes via the wire. If you set "string" (which must be BASE64 encoded), I assume it will decode BASE64 to binary and only then send it (I haven't confirmed this).

So if we want to have the same support in the JSON channel (so switching between the two is seamless), we must also support both "Buffer" and "string" in it.

Or did I misunderstand your comment?

Reply:
the way i understand the bytes field is when using binary grpc channel it doesn't get changed at all. I hope that's true for Buffer or string type. 

then with json the included json_format in python will see any bytes field and base64 encode it when it converts the proto to JSON. This is handled in their formatted so the client that uses it doesn't need to do any special work. so it I think we should just convert any field that is `byte` type (not just named base64 as that is confusing) to base64 for the client in the channel so it matches python (and I assume the other non-js languages). Then all our code examples will be the same, just load an image and set it to the field and our channels will take care of it for you. See if there is a json_format type of thing provided by grpc, there might not as this is already JS.

Reply:
ah yeah if you can't tell if the proto field type is bytes then maybe the "base64" field name is ok for now. I'm not sure if we have any other byte fields currently.